### PR TITLE
Submodules

### PR DESCRIPTION
**Pull request by jimhester**
This still needs some tests, but it works for simple cases, e.g. `install_github("https://github.com/jonkeane/mocapGrip")`

### CODE REVIEW

**Comment by gaborcsardi** [comment]
@jimhester Is this ready?

**Comment by jimhester** [comment]
@gaborcsardi I was adding some tests, it is ready now if you could review it!

**Comment by jimhester** [comment]
Oh I guess the only other thing this could maybe have is a git2r version, right now it relies on having a command line git only. I can add it if you think we need it.

**Comment by gaborcsardi** [comment]
I think it is fine with system git.

File: R/submodule.R [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    x <- readLines(file)
+  }
+
+  # https://git-scm.com/docs/git-config#_syntax
+  # Subsection names are case sensitive and can contain any characters except
+  # newline and the null byte. Doublequote " and backslash can be included by
+  # escaping them as \" and \\
+  double_quoted_string_with_escapes <- '(?:\\\\.|[^"])*'
+
+  sections <- regexpr(
```
Comment (gaborcsardi) [inline R/submodule.R]: I wonder if we should just include a collated rematch2 package, that would simplify things.

File: tests/testthat/test-submodule.R [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  # url = https://foo/bar'
+
+  expect_equal(
+    parse_submodules(z),
+    list()
+  )
+})
+
+test_that("Can install a repo with a submodule", {
+  dir <- tempfile()
```
Comment (gaborcsardi) [inline tests/testthat/test-submodule.R]: Need to skip if no system git.  (If we are skipping the other tests that rely on system git.)

**Comment by gaborcsardi** [comment]
I think it is good now. 

I think if we want to be completely correct we would need to use the `GIT_DIR`  (etc.) env vars. But we'll probably never run into this. So maybe parsing the output of `git  submodule list` etc. is a better approach? 

I  don't think we need to change it now, just sg. to think about next time we write sg for git.

**Comment by jimhester** [comment]
I think the issue with that is the zipball we get from GitHub is not actually a git repository, e.g. it doesn't have a `.git` directory. That is why I can't just run `git submodule init`, `git submodule update` to retrieve the submodules and have to parse them by hand.

**Comment by gaborcsardi** [comment]
Got it. Yeah, these would only matter for local installs. All good then!

**Comment by jimhester** [comment]
Thanks for the review!

**Pull request merged**